manifests/init.pp

```diff
@@ -40,133 +43,153 @@
+        gpgkey         => "file:///etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-${os_maj_release}",
+        descr          => "Extra Packages for Enterprise Linux ${os_maj_release} - \$basearch - Debug",
+        exclude        => $epel_debuginfo_exclude,
+        includepkgs    => $epel_debuginfo_includepkgs,
+      }
     }
 
-    yumrepo { 'epel-source':
-      # lint:ignore:selector_inside_resource
-      mirrorlist     => $epel_source_baseurl ? {
-        'absent' => $epel_source_mirrorlist,
-        default  => 'absent',
-      },
-      # lint:endignore
-      baseurl        => $epel_source_baseurl,
-      failovermethod => $epel_source_failovermethod,
-      proxy          => $epel_source_proxy,
-      enabled        => $epel_source_enabled,
-      gpgcheck       => $epel_source_gpgcheck,
-      gpgkey         => "file:///etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-${os_maj_release}",
-      descr          => "Extra Packages for Enterprise Linux ${os_maj_release} - \$basearch - Source",
-      exclude        => $epel_source_exclude,
-      includepkgs    => $epel_source_includepkgs,
+    if $epel_source_managed {
+      yumrepo { 'epel-source':
+        # lint:ignore:selector_inside_resource
+        mirrorlist     => $epel_source_baseurl ? {
+          'absent' => $epel_source_mirrorlist,
+          default  => 'absent',
+        },
+        # lint:endignore
+        baseurl        => $epel_source_baseurl,
+        failovermethod => $epel_source_failovermethod,
+        proxy          => $epel_source_proxy,
+        enabled        => $epel_source_enabled,
+        gpgcheck       => $epel_source_gpgcheck,
+        gpgkey         => "file:///etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-${os_maj_release}",
+        descr          => "Extra Packages for Enterprise Linux ${os_maj_release} - \$basearch - Source",
+        exclude        => $epel_source_exclude,
+        includepkgs    => $epel_source_includepkgs,
+      }
     }
 
-    file { "/etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-${os_maj_release}":
-      ensure => present,
-      owner  => 'root',
-      group  => 'root',
-      mode   => '0644',
-      source => "puppet:///modules/epel/RPM-GPG-KEY-EPEL-${os_maj_release}",
-    }
+    if $epel_gpg_managed {
+      file { "/etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-${os_maj_release}":
+        ensure => present,
+        owner  => 'root',
+        group  => 'root',
+        mode   => '0644',
+        source => "puppet:///modules/epel/RPM-GPG-KEY-EPEL-${os_maj_release}",
+      }
 
-    epel::rpm_gpg_key{ "EPEL-${os_maj_release}":
-      path   => "/etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-${os_maj_release}",
-      before => Yumrepo['epel','epel-source','epel-debuginfo','epel-testing','epel-testing-source','epel-testing-debuginfo'],
+      epel::rpm_gpg_key{ "EPEL-${os_maj_release}":
+        path   => "/etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-${os_maj_release}",
+        before => Yumrepo['epel','epel-source','epel-debuginfo','epel-testing','epel-testing-source','epel-testing-debuginfo'],
```

Comment:
We could add this condition further up in the init.pp outside of the resource definition. I'm thinking something along the lines of
`Epel::Rpm_gpg_key -> Yumrepo`